app/knights/knight.py

```diff
@@ -0,0 +1,40 @@
+from typing import Union
+
+
+class Knights:
+
+    all_created_knights = {}
+
+    def __init__(
+        self,
+        name: str,
+        power: int,
+        hp: int,
+        armour: list,
+        weapon: dict,
+        potion: Union[dict, None],
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = (
+            sum([i["protection"] if "protection" in i.keys()
+                 else 0 for i in armour])
+            if self.armour
+            else 0
```

Comment:
`.keys()` is redundant, dictionary is automatically iterated through keys
Keys check can be avoided using `.get()` and the default value can be provided when key is not present